### DISCUSSION TITLE
Authenticating Pocketbase in a webhook

### DISCUSSION THREAD

**Discussion by mmkostov** [pocketbase/pocketbase#4144] [Q&A] [answered]
Hi,

I'm trying to figure out a way to authenticate PocketBase in a webhook for a Nextjs project in order to update a user's subscription based on Stripe webhook events. Since the PB client won't work in a webhook route, I'm wondering if it's a good idea to authenticate the request with an admin account? 

```
if (event.type === "customer.subscription.deleted") {
    const subscription = event.data.object as Stripe.Subscription;

    const stripeCustomerId = subscription.customer as string;
    try {
      const authData = await pb.admins.authWithPassword(
        "admin@email.com", //env variable
        "123456789", //env variable
      );

      const user = await pb
        .collection("users")
        .getFirstListItem(`stripeCustomerId="${stripeCustomerId}"`, {
           headers: {
            Authorization: `Bearer ${authData.token}`,
          },
        });

      if (user) {
        await pb.collection("users").update(
          user.id,
          {
            stripeSubscriptionId: null,
            stripePriceId: null,
            stripeCurrentPeriodEnd: null,
            planName: "free", // Set the plan to Free
          },
          {
            headers: {
              Authorization: `Bearer ${authData.token}`,
            },
          }
        );

        console.log("Subscription deleted and user fields reset.");
      } else {
        console.log(`No user found with stripeCustomerId: ${stripeCustomerId}`);
      }
    } catch (error) {
      console.error(`Error querying users in PocketBase: ${error}`);
      return new Response(`Error querying users in PocketBase: ${error}`, {
        status: 500,
      });
    }
  }
```

**Comment by ganigeorgiev** [comment, marked as answer]
> I'm wondering if it's a good idea to authenticate the request with an admin account?

Yes, it should be fine as long as it is executed on the server-side.
To make it slightly more efficient and to avoid calling `authWithPassword` on every execution (_which could be slow because of the bcrypt password verification_), I recommend for such cases to have a global admin SDK instance and reuse it (eg. move it to its own file and then import its exported client). For en example, please check the comment in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130.

**Reply by mmkostov** [reply]
Great! Thanks a lot.
